Source/Core/Resource.js

```diff
@@ -524,6 +589,809 @@ define([
+     * @see {@link http://www.w3.org/TR/cors/|Cross-Origin Resource Sharing}
+     * @see {@link http://wiki.commonjs.org/wiki/Promises/A|CommonJS Promises/A}
+     */
+    Resource.prototype.fetchText = function() {
+        return this.fetch({
+            responseType : 'text'
+        });
+    };
+
+    /**
+     * Creates a Resource from a URL and calls fetchText() on it.
+     *
+     * @param {String} url The url of the resource.
+     * @returns {Promise.<String>|undefined} a promise that will resolve to the requested data when loaded. Returns undefined if <code>request.throttle</code> is true and the request does not have high enough priority.
+     */
+    Resource.fetchText = function (url) {
+        var resource = new Resource(url);
+        return resource.fetchText();
+    };
+
+    // note: &#42;&#47;&#42; below is */* but that ends the comment block early
+    /**
+     * Asynchronously loads the given resource as JSON.  Returns a promise that will resolve to
+     * a JSON object once loaded, or reject if the resource failed to load.  The data is loaded
+     * using XMLHttpRequest, which means that in order to make requests to another origin,
+     * the server must have Cross-Origin Resource Sharing (CORS) headers enabled. This function
+     * adds 'Accept: application/json,&#42;&#47;&#42;;q=0.01' to the request headers, if not
+     * already specified.
+     *
+     * @returns {Promise.<Object>|undefined} a promise that will resolve to the requested data when loaded. Returns undefined if <code>request.throttle</code> is true and the request does not have high enough priority.
+     *
+     *
+     * @example
+     * resource.fetchJson().then(function(jsonData) {
+     *     // Do something with the JSON object
+     * }).otherwise(function(error) {
+     *     // an error occurred
+     * });
+     *
+     * @see {@link http://www.w3.org/TR/cors/|Cross-Origin Resource Sharing}
+     * @see {@link http://wiki.commonjs.org/wiki/Promises/A|CommonJS Promises/A}
+     */
+    Resource.prototype.fetchJson = function() {
+        var promise = this.fetch({
+            responseType : 'text',
+            headers: {
+                Accept : 'application/json,*/*;q=0.01'
+            }
+        });
+
+        if (!defined(promise)) {
+            return undefined;
+        }
+
+        return promise
+            .then(function(value) {
+                if (!defined(value)) {
+                    return;
+                }
+                return JSON.parse(value);
+            });
+    };
+
+    /**
+     * Creates a Resource from a URL and calls fetchJson() on it.
+     *
+     * @param {String} url The url of the resource.
+     * @returns {Promise.<Object>|undefined} a promise that will resolve to the requested data when loaded. Returns undefined if <code>request.throttle</code> is true and the request does not have high enough priority.
+     */
+    Resource.fetchJson = function (url) {
```

Comment:
This can be an object or a url, right?  We should make sure we document these consistently.  My assumption is that the static functions have the same function signature as the constructor, plus any function-specific options, (like post data).  We can talk face to face when you get in to avoid PR churn if that helps.